[`#25`] Added PyVer Crate

Allstreamer opened the pull request:
## Add PyVer crate to project

## Based on:
![image](https://user-images.githubusercontent.com/48365544/184711244-25a7a16b-ef77-464d-9a6d-e1e90111e86e.png)
[PEP-440 Docs](https://peps.python.org/pep-0440/)

### Example Usage
```rust
use pyver::PackageVersion;


let one = PackageVersion::new("v1.0");
let two = PackageVersion::new("2.0");

assert_eq!(true, two > one)
```

#### Commits
- Moved changes from #20 to 25-add-python-version-parsing-pep-440 branch

Allstreamer committed:
Moved changes from #20

- Moved changes from #20 to 25-add-python-version-parsing-pep-440 branch

Allstreamer committed:
Formatted package_version.rs

- Fixed naming
    - `VERSION_VALIDATIOR` to `VERSION_VALIDATOR`
    - `ReleaseCanidate` to `ReleaseCandidate`
- Did some manual formatting
- Ran cargo fmt

Allstreamer committed:
Create CHANGELOG.md

- Created base template changelog including dev branch changes

Allstreamer committed:
Revert "Create CHANGELOG.md"

This reverts commit 3d1a0abb275ecab8d2b13bda50dea059581b5a66.

John15321 commented:
@Allstreamer looks like your package is doing well. Can you change this code to simply use `pyver`?

John15321 committed:
Merge branch 'develop' into 25-add-python-version-parsing-pep-440

Allstreamer commented:
Will do gimme 30-40 min

Allstreamer committed:
Add pyver

- Removed Old dependencies
- Add pyver crate
- Update usages
- Removed package_version

Allstreamer commented:
Should be good to go! @John15321
